client-js/app/elements/expandable-input.ts

```diff
@@ -0,0 +1,96 @@
+import {PolymerElement, html} from "@polymer/polymer";
+import {customElement, property} from "@polymer/decorators";
+
+import {mixinBehaviors} from '@polymer/polymer/lib/legacy/class.js';
+import {IronControlState} from '@polymer/iron-behaviors/iron-control-state.js';
+import "@polymer/paper-input/paper-textarea.js";
+import "@polymer/paper-item/paper-item.js";
+import "@polymer/paper-tooltip/paper-tooltip.js";
+
+/*
+Expandable input. Takes a registry key value and shortens it to fit on the
+screen. When clicked on reveals an editable text box which is shifted into
+focus. Enter fires 'iron-form-submit' which sends the key, value pair as
+event.detail
+*/
+
+@customElement('expandable-input')
+export class ExpandableInput extends mixinBehaviors([IronControlState], PolymerElement) {
+
+  static get template(): HTMLTemplateElement {
+    return html`
+      <style>
+        #long {
+          width: 80%;
+        }
+      </style>
+      <div>
+        <iron-a11y-keys keys="shift+enter" on-keys-pressed="submitForm">
+        <!-- TODO remap newline from enter to "ctrl enter" -->
+          <paper-textarea id="long" value="{{value::input}}" no-label-float hidden></paper-textarea>
+          <paper-item id="short" on-click="selectField" label="{{valueShort}}">{{valueShort}}</paper-item>
+          <paper-tooltip for="long">shift+enter to submit</paper-tooltip>
+        </iron-a11y-keys>
+      </div>
+    `;
+  }
+
+  @property({type: String})
+  name: String;
+
+  @property({type: String})
+  value: String;
+
+  @property({type: Number})
+  cols: Number = 80;
+
+  //holds the shortened value which renders as 'value...' if longer than cols
+  @property({type: String, computed: 'shortenValue(value)'})
+  valueShort: String;
+
+  function() {
+    //This focus-changed event handles the revealing and hiding of the static field
+    this.addEventListener('focused-changed', this.fieldFocus);
+    this.$.long.$.input.$.textarea.addEventListener('keypress', this.keyPressed);
+  }
```

Comment:
expandable-input and some other elements used `this.` frequently in their `expandable-input.html`'s `<script> Polymer({ ... }) </script>` block. In attempting to update, I just copied the use of `this.` from the `<script> Polymer({ ... }) </script>` blocks. However, I am not sure that this copying has led to the correct use of `this.` in the `.ts` files.